<?xml version="1.0"?>
<launch>

    <arg name="enable/publish_image" default="true"/>
    <arg name="enable/publish_markers" default="true"/>
    <arg name="enable/tf" default="true"/>
    <arg name="markers/prefix" default="id_"/>
    <arg name="markers/dictionary" default="0"/>
    <arg name="markers/lengths" default="[15]"/>
    <arg name="camera/tf_name" default="camera_fake"/>
    <arg name="camera/matrix" default="[1130.458671 0.000000 545.323771 0.000000 1154.159372 262.356965 0.000000 0.000000 1.000000]" />
    <arg name="camera/distortion" default="[0.053037 0.116553 0.000778 0.004628 0.000000]" />
    <arg name="markers_param_file" default="$(find marker_localization)/config/markers_config.yaml" />

    <node pkg="marker_localization" type="free_markers_localization" name="free_markers_localization" output="screen">
        <param name="enable/publish_image" value="$(arg enable/publish_image)"/>
        <param name="enable/publish_markers" value="$(arg enable/publish_markers)"/>
        <param name="enable/tf" value="$(arg enable/tf)"/>
        <param name="camera/tf_name" value="$(arg camera/tf_name)"/>
        <param name="markers/prefix" value="$(arg markers/prefix)"/>
        <param name="markers/dictionary" value="$(arg markers/dictionary)"/>
        
        <rosparam param="markers/lengths" subst_value="true">$(arg markers/lengths)</rosparam>
        <rosparam param="camera/matrix" subst_value="true">$(arg camera/matrix)</rosparam>
        <rosparam param="camera/distortion" subst_value="true">$(arg camera/distortion)</rosparam>

        <rosparam command="load" file="$(arg markers_param_file)"></rosparam>

        <remap from="/camera/image_raw" to="/usb_cam/image_raw"/>
        <remap from="detected_markers/image" to="detected_markers/image"/>
        <remap from="detected_markers/pose" to="/detected_markers/output"/>
    </node>

    <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="artag_" args="0.025 0.025 -0.025 0 0 0.707 0.707 camera_fake id_4"/> -->
    <node pkg="tf2_ros" type="static_transform_publisher" name="artag0_" args="0.025 0.025 -0.025 0 0 0.707 0.707 id_4 box"/>
</launch>